### PR TITLE
feat(ESModules): importing for esm modules

### LINKED ISSUE #3992: Incompatible with node ECMAScript Modules
**Please describe the problem you are having in as much detail as possible:**
When using this project with the experimental ECMAScript modules you're not able to destruct imports.
There is a workaround by just destructuring the main export but that is just ugly.

**Include a reproducible code sample here, if possible:**

```js
import { Client } from "discord.js";

const client = new Client();

client.login("TOKEN");
```

**Working example code:**

```js
import Discord from "discord.js";
const { Client } = Discord;

const client = new Client();

client.login("TOKEN");
```

**Further details:**

- discord.js version: 12.0.2
- Node.js version: 13.11.0
- Operating system: Ubuntu 18.04
- Priority this issue should have – please be realistic and elaborate if possible: Low to medium

- [ ] I have also tested the issue on latest master, commit hash:

### PR DESCRIPTION
**Pull request by duncte123**
**Please describe the changes this PR makes and why it should be merged:**
This PR fixes #3992 by adding special exports for esm modules.
The exports in the package.json have been inspired by eris with modifications to work with discord.js

I've built a script that automatically can build the discord.mjs file if needed, the script is as follows:

```javascript
const { writeFileSync } = require('fs');
const Discord = require('discord.js');
const discordExports = Object.keys(Discord);

const template = `import Discord from "../src/index.js";

export default Discord;

export const {
  ${discordExports.join(",\n  ")}
} = Discord;
`;

writeFileSync("./esm/discord.mjs", template);
```

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by chroventer** [comment]
Node.js **fully** provides support for ECMAScript modules as they provide limited interoperability between them and the existing module format, namely, CommonJS. 

Due to the increasing popularity of ECMAScript modules, it is utterly reasonable to allow support for ECMAScript modules. One drawback of the change is that it blocks 'subpath imports' **if paths are not represented in the `exports` field. However, this can, however, be utilized using the subpath, which is already defined in the `exports` field, `./esm`.

```javascript
import submodule from 'discord.js/esm';
```

File: esm/discord.mjs [hunk 1]
```diff
@@ -0,0 +1,91 @@
+'use strict';
+
```
Comment (kyranet) [inline esm/discord.mjs]: ECMAScript Modules are always strict.
```suggestion
```

Comment (duncte123) [reply esm/discord.mjs]: Fixed

**Comment by iCrawl** [comment]
I'd argue this is more `feat` (feature), than a `fix` here.

I assume this is absolutely needed to get modules to work in Node.js?
Pardon my ignorance on that topic.

**Comment by duncte123** [comment]
> I'd argue this is more `feat` (feature), than a `fix` here.
> 
> I assume this is absolutely needed to get modules to work in Node.js?
> Pardon my ignorance on that topic.

Yes this change is required to get modules working in node.js that is why I labeled it as a fix, documentation can be found on this page https://nodejs.org/api/esm.html

A workaround is achievable by using the `esm` package but that is unnecessary blood added to a project.

**Pull request merged**